<?xml version="1.0"?>
<launch>

  <arg name="robot_ns" default="$(env ROBOT_NS)"/>
  <arg name="robot_type"/>
  <arg name="x"/>
  <arg name="y"/>
  <arg name="z"/>
  <arg name="roll"/>
  <arg name="pitch"/>
  <arg name="yaw"/>
  <arg name="width"/>
  <arg name="height"/>
  <arg name="resolution"/>
  <arg name="sim"/>

  <group ns="$(arg robot_ns)">
    <arg name="tf_prefix" value="$(arg robot_ns)_tf"/>
    <param name="tf_prefix" value="$(arg tf_prefix)"/>
    <param name="robot_type" value="$(arg robot_type)"/>
    <include file="$(find costmap_merge)/launch/includes/agent_simulation.launch" if="$(eval arg('sim') == True)">
      <arg name="base" value="$(arg base)"/>
      <arg name="stacks" value="$(arg stacks)"/>
      <arg name="init_pose" value="-x $(arg x) -y $(arg y) -z $(arg z) -R $(arg roll) -P $(arg pitch) -Y $(arg yaw)" />
      <arg name="robot_ns"  value="$(arg robot_ns)"/>
      <arg name="tf_prefix" value="$(arg tf_prefix)"/>
      <arg name="width" value="$(arg width)"/>
      <arg name="height" value="$(arg height)"/>
      <arg name="resolution" value="$(arg resolution)"/>
      <arg name="sim" value="$(arg sim)"/>
    </include>

    <node name="detector" pkg="costmap_merge" type="detector.py" output="screen" if="$(eval arg('robot_type') == 'detector')">
      <rosparam file="$(find costmap_merge)/config/detector.yaml" command="load"/>
    </node>
    <node name="detection_manager" pkg="costmap_merge" type="detection_manager.py" output="screen" if="$(eval arg('robot_type') == 'detector')">
      <rosparam file="$(find costmap_merge)/config/detection_manager.yaml" command="load"/>
    </node>
    <node name="transform_manager" pkg="costmap_merge" type="transform_manager.py" output="screen" if="$(eval arg('robot_type') == 'detector')"/>
    <node name="agent" pkg="costmap_merge" type="agent.py" output="screen"/>
    <node name="costmap_network" pkg="costmap_merge" type="costmap_network.py" output="screen">
      <rosparam file="$(find costmap_merge)/config/costmap_network.yaml" command="load"/>
    </node>
  </group>
 
</launch>
